lldb/source/Plugins/Language/Swift/SwiftLanguage.cpp

```diff
@@ -1761,95 +1762,104 @@ bool SwiftLanguage::GetFunctionDisplayName(
       s << display_name;
       return true;
     }
-    const char *cstr = display_name.data();
-    const char *open_paren = strchr(cstr, '(');
-    const char *close_paren = nullptr;
-    const char *generic = strchr(cstr, '<');
-    // If before the arguments list begins there is a template sign
-    // then scan to the end of the generic args before you try to find
-    // the arguments list.
-    if (generic && open_paren && generic < open_paren) {
-      int generic_depth = 1;
-      ++generic;
-      for (; *generic && generic_depth > 0; generic++) {
-        if (*generic == '<')
-          generic_depth++;
-        if (*generic == '>')
-          generic_depth--;
-      }
-      if (*generic)
-        open_paren = strchr(generic, '(');
-      else
-        open_paren = nullptr;
-    }
-    if (open_paren) {
-      close_paren = strchr(open_paren, ')');
+    return GetFunctionDisplayArgs(args, s, display_name, exe_scope);
+  }
+  }
+  return false;
+}
+
+bool SwiftLanguage::GetFunctionDisplayArgs(
+    VariableList &args, lldb_private::Stream &s, std::string display_name,
+    lldb_private::ExecutionContextScope *exe_scope) {
+  const char *cstr = display_name.data();
+  const char *open_paren = strchr(cstr, '(');
+  const char *close_paren = nullptr;
+  const char *generic = strchr(cstr, '<');
+  // If before the arguments list begins there is a template sign
+  // then scan to the end of the generic args before you try to find
+  // the arguments list.
+  const char *generic_start = generic;
+  if (generic && open_paren && generic < open_paren) {
+    int generic_depth = 1;
+    ++generic;
+    for (; *generic && generic_depth > 0; generic++) {
```

Comment:
Can we get this info from the demangler in the future?
I would not be surprised if at some point in the future a Swift function or type name can contain a `>` character.

Reply:
We could yes, there is already `printFunctionName` in `NodePrinter`. We could add an API that returns the name of the function only.